index.bs

```diff
@@ -2001,6 +2027,68 @@ The [=remote end steps=] with |session| and |command parameters| are:
 
 </div>
 
+#### The session.installWebExtension Command #### {#command-session-installWebExtension}
+
+The <dfn export for=commands>session.installWebExtension</dfn> command installs
+a web-extension for a given session.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+     <pre class="cddl remote-cddl">
+     session.InstallWebExtension = (
+       method: "session.installWebExtension",
+       params: session.InstallWebExtensionParameters
+     )
+
+     session.InstallWebExtensionParameters = {
+       extension: session.Extension
+     }
+     </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+      <pre class="cddl">
+      session.InstallWebExtensionResult = (
+        script.InstallWebExtensionSuccess /
+        script.InstallWebExtensionException
+      )
+
+      session.InstallWebExtensionSuccess = {
+        type: "success",
+        webExtensionId: text
+      }
+
+      session.InstallWebExtensionException = {
+        type: "exception",
+        exceptionDetails: text
+      }
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for session.installWebExtension">
+The [=remote end steps=] with |session| and |command parameters| are:
+
+1. Let the |extension| be the value of the <code>extension</code> field of
+   |command parameters|.
+
+1. [=Try=] to install web-extension with |session| and |extension|.
```

Comment:
So there's quite a few bits missing here:
* It's possible this command isn't supported, so you need to say something like "If installing |extension| isn't supported for any reason return error with error code unsupported operation".
* You need to explictly deal with the different input types, so that end up with some abstract representation of the set of files that make up the extension.
* It would probably be good to make some checks explicit here. For example if there's no `manifest.json` file in the root that should return an error.
* You need to handle the other data passed in e.g. the private browsing mode flag.
* You need a bit more detail on what it means to install a web extension. It's going to be "implementation defined steps", but it should probably describe what the outcome is supposed to be. This will still need to be entirely fallible (i.e. there's a possibility of the install failing for implemenation-specific reasons), as in the current text.